[2.7.0] panic: runtime error: invalid memory address or nil pointer dereference

```
$ git version
git version 2.20.1.windows.1
$ git lfs version
git-lfs/2.7.0 (GitHub; windows amd64; go 1.11.5; git 3b3fa90b)
```

For around a month now, I have been regularly pushing to GitHub and Azure Repos around 25 repos migrated to Git LFS.  I used to use 2.6.1 and I had never noticed such crash.
Yesterday, I updated Git LFS from 2.6.1 to 2.7.0.

While pushing around 25 repos freshly migrated to Git LFS, using 2.7.0, the following error occurred 3-4 times:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x10 pc=0x74f0cf]

goroutine 555 [running]:
github.com/git-lfs/git-lfs/tq.(*basicUploadAdapter).DoTransfer(0xc000826088, 0x0, 0x0, 0xc00084fe00, 0xc0002e7470, 0x0, 0x0, 0x0)
        C:/Users/brian m. carlson/checkouts/git-lfs/tq/basic_upload.go:126 +0x70f
github.com/git-lfs/git-lfs/tq.(*adapterBase).worker(0xc0003e61b0, 0x3, 0x0, 0x0)
        C:/Users/brian m. carlson/checkouts/git-lfs/tq/adapterbase.go:177 +0x54b
created by github.com/git-lfs/git-lfs/tq.(*adapterBase).Begin
        C:/Users/brian m. carlson/checkouts/git-lfs/tq/adapterbase.go:90 +0x2b4
error: failed to push some refs to 'https://XXX'
```

Restarted push completed successfully. So, I don't think this is related to any particular repo or repo size.

I haven't managed to reproduce the crash while running with `GIT_TRACE=1` yet. I will update this report as soon as I get any more details.

--- a/tq/basic_upload.go
+++ b/tq/basic_upload.go
@@ -123,6 +123,12 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 			err = errors.Wrap(err, perr.Error())
 		}
 
+		if res == nil {
+			// We encountered a network or similar error which caused us
+			// to not receive a response at all.
+			return errors.NewRetriableError(err)
+		}
+
 		if res.StatusCode == 429 {
 			retLaterErr := errors.NewRetriableLaterError(err, res.Header["Retry-After"][0])
 			if retLaterErr != nil {

